sdk/src/main/scala/com/horizen/account/forger/AccountForgeMessageBuilder.scala

```diff
@@ -140,9 +141,91 @@ class AccountForgeMessageBuilder(mainchainSynchronizer: MainchainSynchronizer,
     ommersSerializer.toBytes(ommers.asJava).length
   }
 
-  override def getForgingStakeMerklePathInfo(nextConsensusEpochNumber: ConsensusEpochNumber, wallet:  AccountWallet): Seq[ForgingStakeMerklePathInfo] =
-    ???
+  // TODO the stateRoot of the genesis block
+  def getGenesisBlockRootHash =
+    new Array[Byte](32)
 
+  def getStateRoot(history: AccountHistory, nextConsensusEpochNumber: ConsensusEpochNumber) : (ConsensusEpochNumber, Array[Byte]) = {
+
+    // For given epoch N we should get data from the ending of the epoch N-2.
+    // genesis block is the single and the last block of epoch 1 - that is a special case:
+    // Data from epoch 1 is also valid for epoch 2, so for epoch N==2, we should get info from epoch 1.
+    nextConsensusEpochNumber match {
+      case epoch if (epoch <= 2) =>
+        (ConsensusEpochNumber @@ 1, getGenesisBlockRootHash)
+
+      case epoch =>
+
+        val branchPointInfo: BranchPointInfo = getBranchPointInfo(history) match {
+          case Success(info) => info
+          case Failure(ex) =>
+            throw ex
+        }
+
+        // apply two times the api for getting previous epoch last block
+        val parentId_0 = branchPointInfo.branchPointId
```

Comment:
todo: reuse something existing from ConsensusDataProvider

Reply:
Done, reworked a bit an existing method for reusing part of it and creating a new one which is now used here